examples/platform/esp32/shell_extension/heap_trace.cpp

```diff
@@ -0,0 +1,150 @@
+/*
```

Comment:
Why is this under `shell_extension`?

Reply:
We do not locate it under `src/lib/shell` since it's platform specific. The `shell_extension` directory is created for other future custom commands.